### ISSUE TITLE
[cmd/opampsupervisor] Supervisor fails healthcheck with bootstrap config

### ISSUE THREAD

**Issue by BinaryFissionGames** [open-telemetry/opentelemetry-collector-contrib#31897]
### Component(s)

cmd/opampsupervisor

## Description

When running the bootstrap config, the agent doesn't have the healthcheck configuration configured. After bootstrapping, the collector still uses the bootstrap config, which causes the healthcheck to fail every health check interval.

## Steps to Reproduce
Start supervisor. Wait for health check to occur.

## Expected Result
Agent is considered healthy.

## Actual Result
```
2024-03-20T16:18:56.705-0400    ERROR   supervisor/supervisor.go:673    Agent is not healthy    {"error": "Get \"http://localhost:57806\": dial tcp [::1]:57806: connect: connection refused"}
github.com/open-telemetry/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor.(*Supervisor).healthCheck
        /Users/brandonjohnson/git_repos/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor/supervisor.go:673
github.com/open-telemetry/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor.(*Supervisor).runAgentProcess
        /Users/brandonjohnson/git_repos/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor/supervisor.go:733
github.com/open-telemetry/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor.NewSupervisor.func1
        /Users/brandonjohnson/git_repos/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor/supervisor.go:169
```

### Collector version

5bf424dba19b1c4f97957e92b2e23c06a2090673

**Comment by Frapschen** [comment]
It appears that there is a network error. Can you confirm if your agent is exposing the 57806 port?

**Comment by BinaryFissionGames** [comment]
Well, that's exactly it. The agent does not expose that port, but it should, since the config is generated and run by the supervisor.

```yaml
receivers:
  otlp:
    protocols:
      http:
        endpoint: "localhost:58262"
exporters:
  debug:
    verbosity: basic

extensions:
  opamp:
    instance_uid: "01HSERAG6T06AFVGQT5ZYC0GEK"
    server:
      ws:
        endpoint: "ws://localhost:58263/v1/opamp"
        tls:
          insecure: true

service:
  pipelines:
    traces:
      receivers: [otlp]
      exporters: [debug]
  extensions: [opamp]
```

I think the "no-op" config/bootstrap config that the collector runs just needs the health extension config merged into it.

I feel like part of the problem is that the bootstrap config overwrites the collector config (they use the same file), so this is always the initial config for the collector.

Maybe that should be it's own separate issue, though.

**Comment by BinaryFissionGames** [comment]
I believe this ended up fixed in #34159
